File: app/src/main/java/com/kronos/sample/TestActivity.java
package com.kronos.sample;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import com.kronos.router.BindRouter;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by zhangyang on 16/7/16.
 */

//@BindRouter(urls = {"https://www.baidu.com/test"}, weight = 10)
public class TestActivity extends Activity {
    @BindView(R.id.testTv)
    TextView testTv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.new_activity);
        ButterKnife.bind(this);
        try {
            String name = getIntent().getStringExtra("string");
            testTv.setText(name);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: app/src/main/java/com/kronos/sample/TestActivity.kt
package com.kronos.sample

import android.app.Activity
import android.os.Bundle
import com.kronos.router.BindRouter
import com.kronos.sample.databinding.NewActivityBinding

/**
 * Created by zhangyang on 16/7/16.
 */
@BindRouter(urls = ["https://www.baidu.com/test"], weight = 10)
class TestActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = NewActivityBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        val name = intent.getStringExtra("string")
        name?.apply { binding.testTv.text = name }
        binding.resultBtn.setOnClickListener {
            setResult(RESULT_OK)
            finish()
        }

    }
}